src/cmd/cloud.js

```diff
@@ -404,12 +410,18 @@ module.exports = class CloudCommand extends CLICommandBase {
 				return prompts.getCredentials(username, password);
 			})
 			.then(credentials => {
-				const { token, username, password } = credentials;
+				const { token, username, password, sso } = credentials;
 				const msg = 'Sending login details...';
 				const api = new ApiClient();
 
 				this._usernameProvided = username;
 
+				if (sso) {
+					const ssoMessage = 'SSO login in progress...';
+					return this.ui.showBusySpinnerUntilResolved(ssoMessage, ssoLogin())
```

Comment:
This usage of the spinner leaves junk on the first line of output because the function being called prints while the spinner is active.
```
▀ SSO login in progress...
Opening the SSO authorization page in your default browser.
If the browser does not open or you wish to use a different device to authorize this request, open the following URL:

https://id.particle.io/activate?user_code=LNHRQLTL
▌ SSO login in progress...^
```

Because the first request to get a device code is so fast, skip the spinner there. Just have the spinner while waiting for the token to validate.

Reply:
Done here!: https://github.com/particle-iot/particle-cli/pull/637/commits/e7f89fce699d6e006833f35edb8f10722580f522